### PR TITLE
feat: Add support for `vpc_configuration` along with creating an associated security group

### PR DESCRIPTION
**Pull request by mrobinson1022**
## Description
Added a dynamic block to support the vpc_configuration settings and checks to support that block to either create or not create based on the presence of settings in `vpc_configuration = {}`

## Breaking change
This update does require a later version of the aws provider (4.47).  - fix patched in 4.51.0

## Motivation and Context
Required to create a grafana workspace within a vpc.

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes by deploying these changes into my own account
- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Comment by mrobinson1022** [comment]
@bryantbiggs I don't seem to be able to request a review on this PR, if theres anything I can do, please let me know. Thanks!

File: main.tf [hunk 1]
```diff
@@ -26,9 +26,67 @@ resource "aws_grafana_workspace" "this" {
   role_arn                  = var.create_iam_role ? aws_iam_role.this[0].arn : var.iam_role_arn
   stack_set_name            = coalesce(var.stack_set_name, var.name)
 
+  dynamic "vpc_configuration" {
+    for_each = length(var.vpc_configuration) > 0 ? [var.vpc_configuration] : []
+
+    content {
+      security_group_ids = var.create_security_group ? concat([aws_security_group.this[0].id], vpc_configuration.value.security_group_ids) : vpc_configuration.value.security_group_ids
```
Comment (tmszk) [inline main.tf:33]: Is there a reason why `vpc_configuration.value.security_group_ids` is a mandatory field when creating security groups? 

Now it's needed to set an empty list while I would prefer to leave out `security_group_ids` as I intend to to create security groups. So: 

```
  create_security_group = true

  vpc_configuration = {
    subnet_ids = var.subnet_ids
    security_group_ids = []
  }
```

Comment (bryantbiggs) [reply main.tf]: but thank you for the reminder to finish testing this 😅

**Review by bryantbiggs — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 1.7.0 :tada:

File: main.tf (hunk 1 above)
Comment (tmszk) [reply main.tf]: hehe thanks for the PR!  🥇

**Pull request merged**